modules/standard/IO.chpl

```diff
@@ -1579,11 +1579,19 @@ proc open(out error:syserr, path:string="", mode:iomode, hints:iohints=IOHINT_NO
     }
 
     var host = path[host_start..host_end];
-    var port = "";
-    if port_start > 0 then port = path[port_start..port_end];
+    var port_str = "";
+    if port_start > 0 then port_str = path[port_start..port_end];
+
+    var port: int;
+    try! {
+      port = port_str: int;
+    } catch e: IllegalArgumentError {
+      error = ENOENT;
+    }
```

Comment:
@mppf: I'm not confident that this is the right error code to raise if the port_str fails to cast into an int. Any ideas?

Reply:
I think EINVAL just like the others is reasonable.